barretenberg/cpp/src/barretenberg/sumcheck/sumcheck_round.hpp

```diff
@@ -106,6 +137,36 @@ template <typename Flavor> class SumcheckProverRound {
             extended_edge = edge.template extend_to<MAX_PARTIAL_RELATION_LENGTH>();
         }
     }
+    /**
+     * @brief A variant of extend edges used by ZK Sumcheck.
```

Comment:
I'm thinking that this method could also be converged into one that has different shapes enabled by flavor

Reply:
Done. Note: when I'm calling it in the body, strictly speaking, I don't need to add if consexpr(!Flavor::HasZK) and call extend_edges with and without zk_sumcheck_data. But it seems to me, that it is cleaner when !Flavor::HasZK, extend_edges is called without zk_sumcheck_data.